File: Problems/086-partition-list/partition-list.go
package problem086

// ListNode definition.
type ListNode struct {
	Val  int
	Next *ListNode
}

func partition(head *ListNode, x int) *ListNode {
	low := &ListNode{}
	high := &ListNode{}
	lowHead, highHead := low, high

	for head != nil {
		if head.Val < x {
			low.Next = head
			low = low.Next
		} else {
			high.Next = head
			high = high.Next
		}
		head = head.Next
	}

	high.Next = nil
	low.Next = highHead.Next

	return lowHead.Next
}

File: Problems/086-partition-list/partition-list.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    def partition(self, head: ListNode, x: int) -> ListNode:
        low, high = ListNode(0), ListNode(0)
        lowHead, highHead = low, high

        while head:
            if head.val < x:
                low.next = head
                low = low.next
            else:
                high.next = head 
                high = high.next
            head = head.next
        
        high.next = None
        low.next = highHead.next
        
        return lowHead.next